### PR TITLE
added shaker sort in ruby

### LINKED ISSUE #782: shaker sort algorithm in ruby
sir i would like to implement shaker_sort algorithm in ruby language.

### PR DESCRIPTION
**Pull request by AnshitaVishwa**
Fixes #782 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Implemented the Shaker Sort algorithm in Ruby language
#### Languages Used:
- Ruby language
#### Files Added:
- Shaker_Sort.rb

### CODE REVIEW

**Comment by AnshitaVishwa** [comment]
Sir i have implemented the shaker sort algorithm in ruby language.

File: shaker_sort/Shaker_Sort.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        i = 0
+        # If any smaller number is caught
+        for j in i..length-(i+2)
+            # Swap the elements
+            if input[j] > input[j+1]
+              a = input[j]
+              input[j] = input[j+1]
+              input[j+1] = a
+              issort = true
+            end
```
Comment (KanchanThareja) [inline shaker_sort/Shaker_Sort.rb]: fix Indenation to 4 paces

File: shaker_sort/Shaker_Sort.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        # If any smaller number is caught
+        for j in length-(i+2)..i+1
+            # Swap the elements
+            if input[length-i-2] < input[length-i-3]
+              a = input[length-i-2]
+              input[length-i-2] = input[length-i-3]
+              input[length-i-3] = a
+              issort = true
+            end
+        end
```
Comment (KanchanThareja) [inline shaker_sort/Shaker_Sort.rb]: fix here too

File: shaker_sort/Shaker_Sort.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      print(input[i].to_s + " ")
+    end
+end
+input = [2,7,8,1,4,1]
+length = input.length()
+shaker_sort(input , length)
+=begin
+input = [2,7,8,1,4,1]
+output = 1 1 2 4 7 8
+=end
```
Comment (KanchanThareja) [inline shaker_sort/Shaker_Sort.rb]: add a blank line

**Comment by AnshitaVishwa** [comment]
i have done the changes as required. Kindly review it.

**Comment by KanchanThareja** [comment]
@AnshitaVishwa There are some errors in your PR description pls correct them too

**Comment by AnshitaVishwa** [comment]
Ma'am i have done the changes in the PR description .

**Comment by jainaman224** [comment]
@KanchanThareja Is it ready to be merged after squashing commits?

**Comment by KanchanThareja** [comment]
@AnshitaVishwa let me connect with you on slack for your queries.
@jainaman224 Yes its ready to merge.

**Pull request merged**